### PR TITLE
Add c++ helloworld example application

### PR DESCRIPTION
**Pull request by huyhoang8398**
add c++ helloworld example programs

### CODE REVIEW

**Comment by razvand** [comment]
Please use `Add ...` as part of your commit message.

File: lang/c++/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+CC = g++
```
Comment (razvand) [inline lang/c++/Makefile:1]: You should call this `CXX`.

File: lang/c++/Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+CFLAGS  = -g -Wall -fPIC
+LDFLAGS = -static-pie
+
+# The build target 
+TARGET = helloworld
+
+all: $(TARGET)
+
+$(TARGET): $(TARGET).cpp
+	$(CC) $(CFLAGS) -o $(TARGET) $(TARGET).cpp
```
Comment (razvand) [inline lang/c++/Makefile:15]: I think this is implicit and may be removed.

File: lang/c++/Makefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+# compiler flags:
+#  -g     - this flag adds debugging information to the executable file
+#  -Wall  - this flag is used to turn on most compiler warnings
+CFLAGS  = -g -Wall -fPIC
+LDFLAGS = -static-pie
+
+# The build target 
+TARGET = helloworld
+
```
Comment (razvand) [inline lang/c++/Makefile:11]: Make `all` and `clean` targets phony.

```
PHONY: all clean
```

**Review by razvand — CHANGES_REQUESTED** [review]

File: lang/c++/Makefile (hunk 3 above)
Comment (huyhoang8398) [reply lang/c++/Makefile:11]: this should be .PHONY: all clean

File: lang/c++/Makefile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-CC = g++
+CXX = g++
```
Comment (ouss-ch) [inline lang/c++/Makefile:1]: using CXX seems to cause undefined reference errors. I tried CC and it seems to work fine.

**Review by ouss-ch — CHANGES_REQUESTED** [review]

File: lang/c++/helloworld.cpp [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#include <iostream>
+
+int main() {
+  std::cout << "Hello World!";
```
Comment (razvand) [inline lang/c++/helloworld.cpp:4]: Use Tab indentation.

**Review by razvand — CHANGES_REQUESTED** [review]
Thanks, @huyhoang8398 . I made some comments.

Apart from those, please:
* Add the `helloworld` executable file to the commit.
* Remove the `C` folder from the commit.

**Review by razvand — APPROVED** [review]
Thanks, @huyhoang8398.

Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**